backend/addcorpus/json_corpora/import_json.py

```diff
@@ -0,0 +1,282 @@
+    visualize = get_path(field_data, 'options', 'visualize')
+
+    field.language = _parse_language(field_data)
+
+    if searchable and not (sortable or filterable):
+        field.es_mapping = es_mappings.text_mapping()
+        field.display_type = 'text'
+        field.search_filter = {}
+        field.searchable = True
+        if visualize:
+            field.visualizations = [
+                VisualizationType.WORDCLOUD.value
+            ]
+    else:
+        field.es_mapping = es_mappings.keyword_mapping(
+            enable_full_text_search=searchable
+        )
+        field.display_type = 'keyword'
+        if filter_setting == 'show':
+            field.search_filter = {
+                'name': 'MultipleChoiceFilter',
+                'description': f'Select results based on {field.display_name}',
+            }
+        else:
+            field.search_filter = {}
+        field.searchable = searchable
+        field.sortable = sortable
+        if visualize:
+            field.visualizations = [
+                VisualizationType.RESULTS_COUNT.value,
+                VisualizationType.TERM_FREQUENCY.value,
+            ]
+
+    return field
+
+
+def _parse_language(field_data: Dict) -> str:
+    return get_path(field_data, 'language') or ''
+
+
+def _parse_url_field(field: Field, field_data: Dict) -> Field:
+    field.es_mapping = es_mappings.keyword_mapping()
+    field.display_type = 'keyword'
+    field.search_filter = {}
+    return field
+
+
+def _parse_numeric_field(field: Field, field_data: Dict) -> Field:
+    field.display_type = get_path(field_data, 'type')
+
+    if field.display_type == 'integer':
+        field.es_mapping = es_mappings.int_mapping()
+    else:
+        field.es_mapping = es_mappings.float_mapping()
+
+    field.sortable = get_path(field_data, 'options', 'sort')
+
+    filter_setting = get_path(field_data, 'options', 'filter')
+    if filter_setting == 'show':
+        field.search_filter = {
+            'name': 'RangeFilter',
+            'description': f'Select results based on {field.display_name}',
+        }
+    else:
+        field.search_filter = {}
+
+    visualize = get_path(field_data, 'options', 'visualize')
+    if visualize:
+        field.visualizations = [
+            VisualizationType.RESULTS_COUNT.value,
+            VisualizationType.TERM_FREQUENCY.value
+        ]
+        field.visualization_sort = 'key'
+    return field
+
+
+def _parse_date_field(field: Field, field_data: Dict) -> Field:
+    field.display_type = 'date'
+    field.es_mapping = es_mappings.date_mapping()
+    field.sortable = get_path(field_data, 'options', 'sort')
+    filter_setting = get_path(field_data, 'options', 'filter')
+
+    if filter_setting == 'show':
+        field.search_filter = {
+            'name': 'DateFilter',
+            'description': f'Select results based on {field.display_name}',
+        }
+    else:
+        field.search_filter = {}
```

Comment:
Why isn't it possible to read out the min/max date values of the corpus here? Are those optional?

Reply:
They're not specified in the JSON (since it's a simpler version).